Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -354,6 +541,35 @@ bool isFunctionJson(string fullName)
             }
         }
 
+        private async Task LocalAppFetch(IRepository repository, DeploymentInfoBase deploymentInfo, string targetBranch, ILogger logger, ITracer tracer)
+        {
+            var zipDeploymentInfo = (ZipDeploymentInfo)deploymentInfo;
+            var content = !string.IsNullOrEmpty(zipDeploymentInfo.ZipURL)
+                ? await DeploymentHelper.GetZipContentFromURL(zipDeploymentInfo, tracer)
+                : Request.Content;
+            var tempDirPath = repository.RepositoryPath;
+            var targetInfo = FileSystemHelpers.DirectoryInfoFromDirectoryName(tempDirPath);
+            if (targetInfo.Exists)
+            {
+                var moveTarget = Path.Combine(targetInfo.Parent.FullName, Path.GetRandomFileName());
+                using (tracer.Step(string.Format("Renaming extractTargetDirectory({0}) to tempDirectory({1})", targetInfo.FullName, moveTarget)))
```

Comment:
>"Renaming extractTargetDirectory({0}) to tempDirectory({1})", [](start = 49, length = 61)

Why do we do this?

Reply:
We are doing this to move the old content of the temporary directory in a different one to be deleted concurrently along with anything else unnecessary that may be lingering in the folder